lib/Centreon/Object/Relation/Host/Template/Host.php

```diff
@@ -70,6 +70,40 @@ public function insert($fkey, $skey = null)
         $this->db->query($sql, array($fkey, $skey, $order));
     }
 
+    /**
+     * Delete host template / host relation and linked services
+     *
+     * @param int $fkey
+     * @param int $skey
+     * @return void
+     */
+    public function delete($fkey, $skey = null): void
+    {
+        $this->db->beginTransaction();
+        try {
+            parent::delete($fkey, $skey);
+            // Delete linked services :
+            // First get service IDs
+            $linkedServices = $this->getResult(
+                'SELECT service_service_id FROM host_service_relation WHERE host_host_id = ?',
+                [(int)$fkey]
+            );
+            $services = array_map(function($service) {
+                return $service['service_service_id'];
+            }, $linkedServices);
+            if (!empty($services)) {
+                // If there are linked services - delete these services
+                $this->db->query('DELETE FROM host_service_relation 
```

Comment:
you are not using the use recommended by the guideslines.
please use Prepare Statement PHP

https://php.net/manual/fr/pdo.prepared-statements.php